vue/src/util/loki.ts

```diff
@@ -0,0 +1,139 @@
+    const [lb, val] = label.split('=~')
+    let lvalue = val.split(/[""]/)[1]
+    let values = lvalue.split('|')
+    let filtered = values.filter((f) => f.trim() !== value.trim())
+    let lvalues = ''
+    let opr = ''
+    if (filtered.length > 1) {
+      lvalues = filtered.join('|')
+      opr = '=~'
+    } else {
+      lvalues = filtered.join('')
+      opr = '='
+    }
+    const labelmod = lb + opr + '"' + lvalues.trim() + '"'
+    return labelmod
+  },
+  addValueToLabel: (label: string, value: string, isEquals: boolean): string => {
+    const sign = isEquals ? '=' : '=~'
+    const [lb, val] = label.split(sign)
+    const values = val.split(/[""]/)[1]
+    const labelmod = `${lb}=~"${values.trim()} | ${value.trim()}"`
+    return labelmod
+  },
+}
+const parseQuery = {
+  fromLabels: (query: string, keyVal: [string, string], op: string) => {
+    const [key, value] = keyVal
+    const keyValue = `${key}="${value}"`
+    const keySubtValue = `${key}!="${value}"`
+    let queryString = ''
+    let queryArr = parseLog.splitLabels(query)
+    queryArr?.forEach((label) => {
+      const regexQuery = label.match(/([^{}=,~!]+)/gm)
+      const querySplitted = parseLog.splitLabels(query)
+      if (!regexQuery) {
+        return
+      }
+
+      if (
+        !label.includes(key.trim()) &&
+        !label.includes(value.trim()) &&
+        !querySplitted?.some((s) => s.includes(key)) &&
+        !querySplitted?.some((s) => s.includes(key) && s.includes(value))
+      ) {
+        let labelMod = op === '!=' ? keySubtValue : label
+        const parsed = parseLog.addLabel(op, labelMod, keyValue)
+        const regs = parseLog.splitLabels(query).concat(parsed)
+        const joined = regs.join(',')
+        queryString = joined
+      } // add the case for multiple exclusion regex
+      else if (label.includes('=') && label.includes(key) && !label.includes(value)) {
+        let labelMod = parseLog.addValueToLabel(label, value, true)
+        const matches = parseLog.splitLabels(query).join(',').replace(`${label}`, labelMod)
+        queryString = matches
+      } else if (label.includes('=~') && label.includes(key) && label.includes(value)) {
+        const labelMod = parseLog.rmValueFromLabel(label, value)
+        const matches = parseLog.splitLabels(query).join(',').replace(`${label}`, labelMod)
+        queryString = matches
+      } else if (
+        label.includes('=~') &&
+        label.includes(key.trim()) &&
+        !label.includes(value.trim())
+      ) {
+        const labelMod = parseLog.addValueToLabel(label, value, false)
+        queryString = labelMod
+      } else if (
+        !label.includes('=~') &&
+        label.includes(key) &&
+        label.includes(value) &&
+        querySplitted.some((s) => s === label)
+      ) {
+        const filtered = querySplitted.filter((f) => f !== label)
+        const joined = filtered.join(',')
+        queryString = joined
+      }
+    })
+    return queryString
```

Comment:
Looks like this giant if is an ideal case for [early returns](https://medium.com/swlh/return-early-pattern-3d18a41bba8)

Reply:
fixed at refactoring